src/Core/src/Platform/SoftInputExtensions.cs

```diff
@@ -0,0 +1,156 @@
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Maui.Dispatching;
+using System.Threading.Tasks;
+using System.Threading;
+using System;
+
+#if IOS || MACCATALYST
+using PlatformView = UIKit.UIView;
+#elif ANDROID
+using PlatformView = Android.Views.View;
+#elif WINDOWS
+using PlatformView = Microsoft.UI.Xaml.FrameworkElement;
+#elif TIZEN
+using PlatformView = Tizen.NUI.BaseComponents.View;
+#elif (NETSTANDARD || !PLATFORM) || (NET6_0_OR_GREATER && !IOS && !ANDROID && !TIZEN)
+using PlatformView = System.Object;
+using IPlatformViewHandler = Microsoft.Maui.IViewHandler;
+#endif
+
+namespace Microsoft.Maui.Platform;
```

Comment:
Thoughts on 

- Move this extension to `Microsoft.Maui` and up one folder
  - My concern is requiring folks to import `Microsoft.Maui.Platform` to use these APIs since they are
- move the platform level implementations out to other extension methods for those view types
   -  FrameworkElementExtentions (windows)
   - ViewExtentions (iOS)
   - ViewExtentions (Android)